src/scenes/NftDetailPage/NftAdditionalDetails.tsx

```diff
@@ -12,8 +12,16 @@ type Props = {
 };
 
 // The backend converts all token IDs to hexadecimals; here, we convert back
-// TODO: Fix parseInt() for long strings (> 21 characters?), when JS converts to scientific https://gallery.so/robin/21Qt76R6bIyM7fIDsBsgU9DrKCg/21R0nPXR1nlqXP9UD8hn7CQUsUP
-const hexHandler = (str: string) => parseInt(str, 16);
+// https://stackoverflow.com/a/53751162
+const hexHandler = (str: string) => {
+  if (str.length % 2) {
+    str = '0' + str;
+  }
+
+  const bn = BigInt('0x' + str);
```

Comment:
for future readability I think we should use fully written out variable names, that way I can understand what it represents without looking through the rest of the function